transaction/core/src/tx_summary.rs

```diff
@@ -0,0 +1,210 @@
+
+        // Compute the inputs
+        let inputs: Vec<TxInSummary> =
+            zip_exact(tx_prefix.inputs.iter(), pseudo_output_commitments.iter())?
+                .map(|(input, commitment)| {
+                    let mut result = TxInSummary {
+                        pseudo_output_commitment: commitment.clone(),
+                        ..Default::default()
+                    };
+                    if let Some(rules) = &input.input_rules {
+                        result.has_input_rules = true;
+                        let mut associated_keys = rules.associated_tx_target_keys();
+                        input_rules_associated_target_keys.append(&mut associated_keys);
+                    }
+                    result
+                })
+                .collect();
+
+        let outputs: Vec<TxOutSummary> = tx_prefix
+            .outputs
+            .iter()
+            .map(|src| TxOutSummary {
+                masked_amount: src.masked_amount.clone(),
+                target_key: src.target_key,
+                public_key: src.public_key,
+                associated_to_input_rules: input_rules_associated_target_keys
+                    .contains(&src.target_key),
+            })
+            .collect();
+
+        Ok(Self {
+            outputs,
+            inputs,
+            fee: tx_prefix.fee,
+            fee_token_id: tx_prefix.fee_token_id,
+            tombstone_block: tx_prefix.tombstone_block,
+        })
+    }
+}
+
+/// A subset of the data of a TxOut.
+///
+/// Fog hint and memo are omitted to reduce size and complexity on HW device,
+/// which can't really do much with those and isn't very interested in them
+/// anyways.
+#[derive(Clone, Deserialize, Digestible, Eq, Hash, Message, PartialEq, Serialize, Zeroize)]
+pub struct TxOutSummary {
+    /// The amount being sent.
+    #[prost(oneof = "MaskedAmount", tags = "1, 2")]
+    pub masked_amount: Option<MaskedAmount>,
+
+    /// The one-time public address of this output.
+    #[prost(message, required, tag = "3")]
+    pub target_key: CompressedRistrettoPublic,
+
+    /// The per output tx public key
+    #[prost(message, required, tag = "4")]
+    pub public_key: CompressedRistrettoPublic,
+
+    /// Whether or not this output is associated to an input with rules
+    #[prost(bool, tag = "5")]
+    pub associated_to_input_rules: bool,
+}
+
+/// Data in a TxSummary associated to a transaction input.
+///
+/// This includes only the pseudo output commitment and the InputRules if any,
+/// omitting the Ring and the proofs of membership.
+#[derive(Clone, Deserialize, Digestible, Eq, Hash, Message, PartialEq, Serialize, Zeroize)]
+pub struct TxInSummary {
+    /// Commitment of value equal to the real input.
+    #[prost(message, required, tag = "1")]
+    pub pseudo_output_commitment: CompressedCommitment,
+
+    /// Whether there are input rules associated to this input
+    #[prost(bool, tag = "2")]
+    pub has_input_rules: bool,
```

Comment:
:question: Similar to above what value does this flag provide in the summary?

Reply:
it tells the device that this input does not come from us (the signer), but rather from some counterparty, so it should not be reflected as deducting from our balance, but rather someone else's.

Reply:
I'm sure I'm missing some of the bigger picture, and I'm not fully aware of the threat model (let alone lacking experience to point out weaknesses reliably)
Let us say the application builds up a Tx:
1. 10mob this client that is willing to trade (no input rules)
2. 10 eusd for the trade (input rules from partial fill originator)
3. 200mob (input rules) but this is coming from this client's wallet  

Could 3 happen?  The application is attempting to transfer 200mob from the client to some address, with some arbitrary input rule that can be met.  Would the HW wallet hide this 200mob transaction from the client.

It seems like input rules are designed to evolve over time.  If this is the case, keying off `has_input_rules` may quickly turn into a misnomer.  What if someone decides to add a date/time expirey via input rules.  Then the input rules could be for this client and not a counterparty.

Reply:
these are great questions -- lets try to move the discussion to MCIP 52, i still haven't actually written out what I'm trying to do and what my assumptions are. it's possible I've messed up, but let me take some time to write why I think this rev is okay